questionnaire/section/section.test.js

```diff
@@ -0,0 +1,220 @@
+                        sectionDefinition: {
+                            schema: {
+                                type: 'object',
+                                properties: {
+                                    'q-applicant-when-did-the-crime-start': {
+                                        title: 'When did it start?',
+                                        meta: {
+                                            keywords: {
+                                                format: {
+                                                    precision: 'YYYY-MM',
+                                                    defaults: {
+                                                        DD: '01'
+                                                    }
+                                                }
+                                            }
+                                        },
+                                        type: 'string',
+                                        format: 'date-time',
+                                        description:
+                                            'For example, 02 2020. You can enter an approximate date.'
+                                    }
+                                }
+                            }
+                        }
+                    });
+                    const data = {
+                        'q-applicant-when-did-the-crime-start': '2021-01-01T00:00:00.000Z'
+                    };
+                    const sectionAttributes = section.getAttributesByData(data);
+
+                    expect(sectionAttributes).toEqual([
+                        {
+                            id: 'q-applicant-when-did-the-crime-start',
+                            type: 'simple',
+                            label: 'When did it start?',
+                            value: '2021-01-01T00:00:00.000Z',
+                            format: {
+                                value: 'date-time',
+                                precision: 'YYYY-MM',
+                                defaults: {
+                                    DD: '01'
+                                }
+                            }
+                        }
+                    ]);
+                });
+
+                describe('Given the supplied data has no corresponding keys in the section schema', () => {
+                    it('should return an empty list of attributes', async () => {
+                        const section = createSection({
+                            sectionDefinition: fixtures.sectionWithSimpleAttributeSingleValue
+                        });
+                        const data = {
+                            'this-key-does-not-exist-in-the-section-schema': 'dummy value'
+                        };
+                        const sectionAttributes = section.getAttributesByData(data);
+
+                        expect(sectionAttributes).toEqual([]);
```

Comment:
Would it be worth having a test for a data object that contains at least one genuine answer and 1 dummy one?

Reply:
I reckon it's ok. There is test for ignoring non-existent keys and another for returning keys that exist, and a another that returns 2 out of 3 things. Pretty much everything is covered in the coverage report. I can add one if you like, but I don't think it's going to increase coverage